modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/resource/v1_0/TaxonomyVocabularyResourceImpl.java

```diff
@@ -981,12 +981,18 @@ private AssetVocabulary _updateVocabulary(
 				_getAssetLibraryGroupIds(taxonomyVocabulary));
 		}
 
+		boolean internalVisibilityType =
+			TaxonomyVocabulary.VisibilityType.INTERNAL.equals(
+				taxonomyVocabulary.getVisibilityType());
+
 		return _assetVocabularyService.updateVocabulary(
-			assetVocabulary.getVocabularyId(), null, titleMap, descriptionMap,
+			assetVocabulary.getVocabularyId(), taxonomyVocabulary.getName(),
+			titleMap, descriptionMap,
 			_getSettings(
 				taxonomyVocabulary.getAssetTypes(),
 				assetVocabulary.getGroupId(),
 				taxonomyVocabulary.getMultiValued()),
+			internalVisibilityType ? 1 : 0,
```

Comment:
@joshuacords Can you use the visiblity constants in `AssetVocabularyConstants` here?